### PR TITLE
initial template of lazyrelin pass

### PR DESCRIPTION
**Pull request by ahmedshakill**
Implementation of #599 

Here is the current progress status
- [x] The lazy relin pass is visible in heir-opt. 
- [x] Looked into google or-tools tutorial on LP and MIP. 
- [x] Need an in-depth look into Jeremy's ILP blog.  
- [ ] Translate the minimization problem into an ILP one. 
- [ ] Introduce trait/interface

### CODE REVIEW

**Review by asraa — COMMENTED** [review]
Hey @ahmedshakill ! It looks like you're on the right track (love that you're following the blog), and you've set up the pass boilerplate and the analysis class up. From here you'll need to implement the logic in the solver and then call it in the pass.

If you need some more help, feel free to reach out on discord too!

**Comment by asraa** [comment]
Hey! @ahmedshakill what's the state of this? Do you need me to take a review of it?

**Comment by ahmedshakill** [comment]
Hi, @asraa . For analysis I'm somewhat familiar with linear programming concepts. I looked at jeremy's blogs.
Currently, to play around, I'm trying to insert a relin op when I face a mul op. But the insertion fails with something like
`from_basis information is required`.
So I'm looking at how to properly create a relin op using OpBuilder. I'll need help with formulating the model for analysis. 
I'll come up with questions on discord, I guess.

File: WORKSPACE [hunk 1]
```diff
@@ -352,3 +352,9 @@ git_repository(
     patches = ["@heir//bazel/openfhe:add_config_core.patch"],
     remote = "https://github.com/openfheorg/openfhe-development.git",
 )
+git_repository(
```
Comment (j2kun) [inline WORKSPACE:355]: This dependency is already defined on line 325, fyi

**Comment by AlexanderViand-Intel** [comment]
See #1016 for a continuation of this PR

**Pull request closed without merging**